clients/src/main/java/org/apache/kafka/common/PartitionInfo.java

```diff
@@ -88,6 +90,28 @@ public Node[] offlineReplicas() {
         return offlineReplicas;
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(topic, partition, leader, replicas, inSyncReplicas, offlineReplicas);
```

Comment:
Should we use `Arrays.hashCode(replicas)` to make sure the hash code is generated by "content" rather than reference?

Reply:
Thanks for the comment! From the docs it looks like `Objects.hash()` delegates to `Arrays.hashCode()` (when we're passing more than one Object) so these should be equivalent. 
```
   /**
    * Generates a hash code for a sequence of input values. The hash
    * code is generated as if all the input values were placed into an
    * array, and that array were hashed by calling {@link
    * Arrays#hashCode(Object[])}.
    * 
    * ...
    * ...
    * <b>Warning: When a single object reference is supplied, the returned
    * value does not equal the hash code of that object reference.</b> This
    * value can be computed by calling {@link #hashCode(Object)}.
    *
    * @param values the values to be hashed
    * @return a hash value of the sequence of input values
    * @see Arrays#hashCode(Object[])
    * @see List#hashCode
    */
    public static int hash(Object... values) {
        return Arrays.hashCode(values);
    }
```

Reply:
I think I understand what you're saying, even though `Arrays.hashCode()` is called on the params, internally it computes the hash by calling `element.hashCode()` on each param (`object#hashCode`). If this were called on an int array for instance, this would result in different hashes being produced no matter the equality of their contents. However, since this is called on our Node array which has an overridden hashCode, we get the expected results still.

I see how 
```
Objects.hash(topic, partition, leader, Arrays.hashCode(replicas), Arrays.hashCode(inSyncReplicas), Arrays.hashCode(offlineReplicas));
```
is technically more correct/explicit, I'll make the change.